stdlib/grpc/src/main/ballerina/src/grpc/service_endpoint.bal

```diff
@@ -94,6 +94,46 @@ public type Listener object {
     }
 };
 
+# The stream iterator object that is used to iterate through the stream messages.
+#
+type StreamIterator object {
+    private boolean isClosed = false;
+
+    public function next() returns record {|anydata value;|}|error? {
+        if (self.isClosed) {
+            return StreamEndError (message = "Stream is closed. Therefore, "
+                           + "no operations are allowed further on the stream.");
+        }
+        anydata|handle |error? result = nextResult(self);
+        if (result is anydata) {
+            if (result is ()) {
+                self.isClosed = true;
+                return prepareError (EOS_REASON, "End of stream reached", result);
+            }
+            return {value: result};
+        } else if (result is handle) {
+            return {value: java:toString(result)};
+        } else {
+            return result;
+        }
+    }
+
+    public function close() returns error? {
+        if (!self.isClosed) {
+            self.isClosed = true;
+            error? result = closeStream(self);
+            if (result is error) {
+                return result;
+            } else {
+                return prepareError (EOS_REASON, "Stream closed from server side", result);
+            }
```

Comment:
```suggestion
            return closeStream(self);
```

Reply:
If we get `()`, means we closed successfully

Reply:
I made the change